src/3d/qgs3dutils.cpp

```diff
@@ -442,6 +442,15 @@ QgsVector3D Qgs3DUtils::worldToMapCoordinates( const QgsVector3D &worldCoords, c
                       worldCoords.y() + origin.z() );
 }
 
+QgsVector3D Qgs3DUtils::transformCoordinates( const QgsVector3D &coords, QgsCoordinateTransform coordTrans )
```

Comment:
This method actually could be moved to directly to QgsCoordinateTransform as another version of the transform() member function - so that it could be possibly used elsewhere too:
```c++
QgsVector3D transform( const QgsVector3D &coords ) const;
```